File: C++ Programming/Stack.cpp
//stack
#include<iostream>
using namespace std;

typedef struct node
{
    int data;
    struct node * next;
}NODE ,*PNODE;
/*
typedef struct node NODE
typedef struct node* PNODE
*/
class Stack
{
    private:
        PNODE first;
        
        int Count;
        
    public: 
        Stack();
        void Push(int No);
        void Pop();
        void Display();
        int CountNode();
};
Stack :: Stack()
{
    first = NULL;
    
    Count =0;
}
void Stack :: Push(int no)
{
    PNODE newn =NULL;
    newn=new(NODE);
    newn->data=no;
    newn->next=NULL;
   

    if(first==NULL)
    {
        first=newn;
    }
    else
    {
        newn->next=first;
        first=newn;
    }
    Count++;
}

void Stack::Pop()
{
    if(first == NULL)
    {
        return;
    }
    else if(first->next == NULL)
    {
        delete first;
        first = NULL;
    }
    else
    {
        PNODE temp = first;

        first = first->next;
        delete temp;
    }
    Count--;
}
void Stack::Display()
{
    PNODE temp = first;

    cout<<"Elements of the linked list are : \n";
    
    while(temp != NULL) // Type 1
    {
        cout<<"| "<<temp->data<<" | -> ";
        temp = temp ->next;
    }
    cout<<"NULL \n";
}

int Stack::CountNode()
{
    return Count;
}
int main()
{
    Stack obj;
    int iRet = 0;

    obj.Push(111);
    obj.Push(101);
    obj.Push(51);
    obj.Push(21);
    obj.Push(11);

    
    obj.Display();

    iRet = obj.CountNode();
    cout<<"Number of elements in the linked list are : "<<iRet<<"\n";

    obj.Pop();
    obj.Display();

    iRet = obj.CountNode();
    cout<<"Number of elements in the linked list are : "<<iRet<<"\n";


    
    return 0;
}

File: Java Programming/Stack.java

//Stack
class Stack
{
    public static void main(String arg[])
    {
        int iRet = 0;

        StackCode obj = new StackCode();

        obj.Push(51);
        obj.Push(21);
        obj.Push(11);

        obj.Display();

        iRet = obj.Count();
        System.out.println("Number of elements are : "+iRet);

        obj.Pop();
        obj.Display();

        iRet = obj.Count();
        System.out.println("Number of elements are : "+iRet);

    }
}


class node
{
    public int data;
    public node next;

    public node(int no)
    {
        data = no;
        next = null;
    }
}
class StackCode
{
    private node first;
    private int Count;

    public StackCode()
    {
        first = null;
        Count = 0;
    }

    public void Push(int no)
    {
        node newn = new node(no);

        if(first == null)   // LL is empty
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
        Count++;
    }
    public void Pop()
    {
        if(first == null)       // LL is empty
        {
            return;
        }
        else if(first.next == null) // LL contains single node
        {
            first = null;
        }
        else        // LL contains more than one node
        {
            first = first.next;
        }
        Count--;
    }
    public void Display()
    {
        node temp = first;

        System.out.println("Elements of the Linked List are : ");

        while(temp != null)
        {
                System.out.print("| "+ temp.data +" |->");
                temp = temp.next;
        }
        System.out.println("null");
    }

    public int Count()
    {
        return Count;
    }
}
